pkg/kubefedctl/orphaning/orphaning.go

```diff
@@ -0,0 +1,141 @@
+)
+
+type orphanResource struct {
+	options.GlobalSubcommandOptions
+	typeName          string
+	resourceName      string
+	resourceNamespace string
+}
+
+// Bind adds the join specific arguments to the flagset passed in as an argument.
+func (o *orphanResource) Bind(flags *pflag.FlagSet) error {
+	flags.StringVarP(&o.resourceNamespace, "namespace", "n", "", "If present, the namespace scope for this CLI request")
+	err := flags.MarkHidden("kubefed-namespace")
+	if err != nil {
+		return err
+	}
+	err = flags.MarkHidden("dry-run")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// NewCmdOrphaning the head of orphaning-deletion sub commands
+func NewCmdOrphaning(cmdOut io.Writer, config util.FedConfig) *cobra.Command {
+	//	opts := &orphanResource{}
+	cmd := &cobra.Command{
+		Use:   "orphaning-deletion",
+		Short: "Manage orphaning delete policy",
+		Long:  "Manage orphaning delete policy",
+		Run: func(cmd *cobra.Command, args []string) {
+			err := cmd.Help()
+			if err != nil {
+				klog.Fatalf("Error: %v", err)
+			}
+		},
+	}
+	cmd.AddCommand(newCmdEnableOrphaning(cmdOut, config))
+	cmd.AddCommand(newCmdDisableOrphaning(cmdOut, config))
+	cmd.AddCommand(newCmdStatusOrphaning(cmdOut, config))
+
+	return cmd
+}
+
+//  Complete ensures that options are valid and marshals them if necessary.
+func (o *orphanResource) Complete(args []string, config util.FedConfig) error {
+
+	if len(args) == 0 {
+		return errors.New("resource type is required")
+	}
+
+	o.typeName = args[0]
+
+	if len(args) == 1 {
+		return errors.New("resource name is required")
+	}
+	o.resourceName = args[1]
+
+	return o.SetNamespace(config)
+}
+
+// If namespace was not set as an argument, it will be the current context namespace
+func (o *orphanResource) SetNamespace(config util.FedConfig) error {
+
+	if o.resourceNamespace != "" {
+		return nil
+	}
+	ns, _, err := config.GetClientConfig(o.HostClusterContext, o.Kubeconfig).Namespace()
+	if err != nil {
+		return errors.Wrapf(err, "Failed to get ClientConfig for host cluster context '%s' and kubeconfig '%s'", o.HostClusterContext, o.Kubeconfig)
+	}
+
+	if ns == "" {
+		ns = "default"
+	}
+	o.resourceNamespace = ns
+	return nil
+}
+
+// Returns the Federated resource where the orphaning-deletion will be managed, and a corresponding Resources Interface
+func (o *orphanResource) GetFederatedResource(config util.FedConfig) (*unstructured.Unstructured, dynamic.ResourceInterface, error) {
+	hostConfig, err := config.HostConfig(o.HostClusterContext, o.Kubeconfig)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "Failed to get host cluster config")
+	}
+	// Lookup kubernetes API availability
+	apiResource, err := enable.LookupAPIResource(hostConfig, o.typeName, "")
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "Failed to find target API resource %s", o.typeName)
+	}
+	klog.V(2).Infof("API Resource for %s found", o.typeName)
+	if !util.IsFederatedAPIResource(apiResource.Kind, apiResource.Group) {
```

Comment:
The point of the orphaning commands is to add/remove/check for an annotation.  Why does the command need to be restricted to a weak check for whether the resource is federated vs just applying/removing the annotation on any resource?

Reply:
I think we can mislead users if we allow applying/removing the annotation on any resource. In the end of the day, a user wants to change behavior of a federated resource deletion, and not to add or remove the annotation. If a wrong resource was provided the command should fail. 
What do you think ?

Reply:
If `IsFederatedAPIResource` was guaranteed to identify a federated resource I would agree with you that preventing addition of the annotation would be desirable, but as per previous comments that is not the case. How about adding a `--force` argument to `enable` and `disable` commands to ensure that a user can still use the command even if the check fails? And have the `status` command always indicate orphaning (rather than requiring `--force` for a read-only operation)?